core/state/dump.go

```diff
@@ -127,18 +143,33 @@ func (self *StateDB) dump(c collector, excludeCode, excludeStorage, excludeMissi
 			}
 		}
 		c.onAccount(addr, account)
+
+		if maxResults <= 0 {
+			continue
+		}
+
+		count++
+		if count >= maxResults {
```

Comment:
```suggestion
		if maxresults > 0 && count >= maxResults {
```
.. and remove the `if` a few lines up